Panic in incrementHumioClusterPodRevision

Seeing a pod crash in humio-operator:0.13.0. Seems to proceed OK after pod restart, so not a huge issue.

```
E1210 17:32:23.059421       1 runtime.go:78] Observed a panic: "assignment to entry in nil map" (assignment to entry in nil map)
goroutine 817 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x1814b80, 0x1bf7530)
	/go/pkg/mod/k8s.io/apimachinery@v0.21.3/pkg/util/runtime/runtime.go:74 +0x95
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.21.3/pkg/util/runtime/runtime.go:48 +0x86
panic(0x1814b80, 0x1bf7530)
	/usr/local/go/src/runtime/panic.go:965 +0x1b9
github.com/humio/humio-operator/controllers.(*HumioClusterReconciler).incrementHumioClusterPodRevision.func1(0xc002712c30, 0x21)
	/workspace/controllers/humiocluster_annotations.go:58 +0x14a
k8s.io/client-go/util/retry.OnError.func1(0x0, 0x0, 0x0)
	/go/pkg/mod/k8s.io/client-go@v0.21.3/util/retry/util.go:51 +0x3c
k8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtection(0xc0011f4b28, 0x1667000, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.21.3/pkg/util/wait/wait.go:211 +0x69
k8s.io/apimachinery/pkg/util/wait.ExponentialBackoff(0x989680, 0x3ff0000000000000, 0x3fb999999999999a, 0x5, 0x0, 0xc0011f4b28, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.21.3/pkg/util/wait/wait.go:399 +0x55
k8s.io/client-go/util/retry.OnError(0x989680, 0x3ff0000000000000, 0x3fb999999999999a, 0x5, 0x0, 0x1acd3a0, 0xc0011f4c00, 0x3ff0000000000000, 0x3fb999999999999a)
	/go/pkg/mod/k8s.io/client-go@v0.21.3/util/retry/util.go:50 +0xa6
k8s.io/client-go/util/retry.RetryOnConflict(...)
	/go/pkg/mod/k8s.io/client-go@v0.21.3/util/retry/util.go:104
github.com/humio/humio-operator/controllers.(*HumioClusterReconciler).incrementHumioClusterPodRevision(0xc0002194f0, 0x1c445f8, 0xc001047680, 0xc0027a0580, 0x1a055ba, 0x7, 0x0, 0x0, 0x1a00d60)
	/workspace/controllers/humiocluster_annotations.go:51 +0x210
github.com/humio/humio-operator/controllers.(*HumioClusterReconciler).ensureMismatchedPodsAreDeleted(0xc0002194f0, 0x1c445f8, 0xc001047680, 0xc0027a0580, 0x0, 0x0, 0xd, 0x1c67da8)
	/workspace/controllers/humiocluster_controller.go:1912 +0x132f
github.com/humio/humio-operator/controllers.(*HumioClusterReconciler).Reconcile(0xc0002194f0, 0x1c445f8, 0xc001047680, 0xc0020d7815, 0x5, 0xc0020d78c0, 0xd, 0xc001047600, 0x0, 0x0, ...)
	/workspace/controllers/humiocluster_controller.go:138 +0xa2b
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0003f1180, 0x1c44550, 0xc00041a000, 0x1873ae0, 0xc000dc6b00)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.2/pkg/internal/controller/controller.go:298 +0x30d
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0003f1180, 0x1c44550, 0xc00041a000, 0x0)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.2/pkg/internal/controller/controller.go:253 +0x205
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2(0xc0003fe010, 0xc0003f1180, 0x1c44550, 0xc00041a000)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.2/pkg/internal/controller/controller.go:214 +0x6b
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.2/pkg/internal/controller/controller.go:210 +0x425
panic: assignment to entry in nil map [recovered]
	panic: assignment to entry in nil map

goroutine 817 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.21.3/pkg/util/runtime/runtime.go:55 +0x109
panic(0x1814b80, 0x1bf7530)
	/usr/local/go/src/runtime/panic.go:965 +0x1b9
github.com/humio/humio-operator/controllers.(*HumioClusterReconciler).incrementHumioClusterPodRevision.func1(0xc002712c30, 0x21)
	/workspace/controllers/humiocluster_annotations.go:58 +0x14a
k8s.io/client-go/util/retry.OnError.func1(0x0, 0x0, 0x0)
	/go/pkg/mod/k8s.io/client-go@v0.21.3/util/retry/util.go:51 +0x3c
k8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtection(0xc0011f4b28, 0x1667000, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.21.3/pkg/util/wait/wait.go:211 +0x69
k8s.io/apimachinery/pkg/util/wait.ExponentialBackoff(0x989680, 0x3ff0000000000000, 0x3fb999999999999a, 0x5, 0x0, 0xc0011f4b28, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.21.3/pkg/util/wait/wait.go:399 +0x55
k8s.io/client-go/util/retry.OnError(0x989680, 0x3ff0000000000000, 0x3fb999999999999a, 0x5, 0x0, 0x1acd3a0, 0xc0011f4c00, 0x3ff0000000000000, 0x3fb999999999999a)
	/go/pkg/mod/k8s.io/client-go@v0.21.3/util/retry/util.go:50 +0xa6
k8s.io/client-go/util/retry.RetryOnConflict(...)
	/go/pkg/mod/k8s.io/client-go@v0.21.3/util/retry/util.go:104
github.com/humio/humio-operator/controllers.(*HumioClusterReconciler).incrementHumioClusterPodRevision(0xc0002194f0, 0x1c445f8, 0xc001047680, 0xc0027a0580, 0x1a055ba, 0x7, 0x0, 0x0, 0x1a00d60)
	/workspace/controllers/humiocluster_annotations.go:51 +0x210
github.com/humio/humio-operator/controllers.(*HumioClusterReconciler).ensureMismatchedPodsAreDeleted(0xc0002194f0, 0x1c445f8, 0xc001047680, 0xc0027a0580, 0x0, 0x0, 0xd, 0x1c67da8)
	/workspace/controllers/humiocluster_controller.go:1912 +0x132f
github.com/humio/humio-operator/controllers.(*HumioClusterReconciler).Reconcile(0xc0002194f0, 0x1c445f8, 0xc001047680, 0xc0020d7815, 0x5, 0xc0020d78c0, 0xd, 0xc001047600, 0x0, 0x0, ...)
	/workspace/controllers/humiocluster_controller.go:138 +0xa2b
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0003f1180, 0x1c44550, 0xc00041a000, 0x1873ae0, 0xc000dc6b00)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.2/pkg/internal/controller/controller.go:298 +0x30d
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0003f1180, 0x1c44550, 0xc00041a000, 0x0)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.2/pkg/internal/controller/controller.go:253 +0x205
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2(0xc0003fe010, 0xc0003f1180, 0x1c44550, 0xc00041a000)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.2/pkg/internal/controller/controller.go:214 +0x6b
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.2/pkg/internal/controller/controller.go:210 +0x425
```

--- a/controllers/humiocluster_annotations.go
+++ b/controllers/humiocluster_annotations.go
@@ -52,6 +52,9 @@ func (r *HumioClusterReconciler) incrementHumioClusterPodRevision(ctx context.Co
 				return err
 			}
 		}
+		if hc.Annotations == nil {
+			hc.Annotations = map[string]string{}
+		}
 		hc.Annotations[revisionKey] = strconv.Itoa(revisionValue)
 		r.setRestartPolicy(hc, restartPolicy)
 		return r.Update(ctx, hc)

